packages/flutter/lib/src/material/bottom_app_bar.dart

```diff
@@ -155,9 +155,6 @@ class _BottomAppBarClipper extends CustomClipper<Path> {
 
   @override
   bool shouldReclip(CustomClipper<Path> oldClipper) {
-    if (oldClipper.runtimeType != _BottomAppBarClipper)
-      return true;
-
     final _BottomAppBarClipper typedOldClipper = oldClipper;
```

Comment:
that would also let you get rid of this line, and could indeed turn the whole method into a one-liner.

Reply:
done